### DISCUSSION TITLE
Why is the lower accuracy model saved as best?

### DISCUSSION THREAD

**Discussion by ep0p** [PaddlePaddle/PaddleOCR#14055] [Q&A] [answered]
I am training the PPOCRv3 model for text recognition and noticed something unexpected. During training, I consistently saw accuracies above 0.92, for example

[2024/10/17 21:24:02] ppocr INFO: epoch: [10/10], global_step: 62540, lr: 0.000501, acc: 0.937500 ....
[2024/10/17 21:24:23] ppocr INFO: epoch: [10/10], global_step: 62550, lr: 0.000501, acc: 0.925781 ....
[2024/10/17 21:24:37] ppocr INFO: epoch: [10/10], global_step: 62560, lr: 0.000500, acc: 0.941406 ....

However, after training, the model saved as "best" has a lower accuracy of 0.87:
[2024/10/17 21:24:41] ppocr INFO: save model in ./output/v3_latin_mobile_noisy_10epochs/latest
[2024/10/17 21:24:42] ppocr INFO: best metric, acc: 0.8671828170961855, is_float16: False, norm_edit_dis: 0.9287352828945985, fps: 3108.9107400895145, best_epoch: 3

Why was the model with lower accuracy (0.87) saved instead of the ones with higher accuracy (above 0.92) during training?

**Comment by GreatV** [comment, marked as answer]
The higher training accuracies you observed (above 0.92) were calculated on the training data and may be inflated due to overfitting. The "best" model is saved based on the highest validation accuracy achieved on a separate validation set. In your case, the model achieved its peak validation accuracy of 0.87 at epoch 3, so the model from that epoch was saved as the "best," despite higher training accuracies in later epochs.

**Comment by ep0p** [comment]
Thanks a lot for your reply!

I have one more question regarding the model files generated during training. When the model is saved, I notice multiple types of .pdparams files (except the iterative ones).

1. best_accuracy
2. latest
3. best_model/best_model.pdparams

I understand that the export is typically done using latest.pdparams, which contains the model parameters from the most recent epoch. However, I am curious about the others. Could you explain the difference between these files and when I should use one instead of the others?

**Reply by GreatV** [reply]
Use the model in the `best_accuracy` directory for export.

**Reply by GreatV** [reply]
For details, visit: https://paddlepaddle.github.io/PaddleOCR/latest/en/ppocr/model_train/detection.html#4-inference

**Comment by ep0p** [comment]
got it, thanks a lot
